pkg/rancher-desktop/config/settings.ts

```diff
@@ -317,39 +352,78 @@ export function updateFromCommandLine(cfg: Settings, commandLineArgs: string[]):
 
   return cfg;
 }
+
 /**
  * Load the settings file or create it if not present.  If the settings have
  * already been loaded, return it without re-loading from disk.
  */
 export function load(): Settings {
+  let deploymentProfiles: DeploymentProfileType = { defaults: {}, locked: {} };
+  let setDefaultMemory = false;
+  let lockedProfileSettings: RecursivePartial<Settings>;
+
+  try {
+    deploymentProfiles = readDeploymentProfiles();
+    lockedProfileSettings = deploymentProfiles.locked;
+  } catch {
+    lockedProfileSettings = {};
+  }
   try {
     settings ??= loadFromDisk();
   } catch (err: any) {
     settings = clone(defaultSettings);
     if (err.code === 'ENOENT') {
-      _isFirstRun = true;
-      if (os.platform() === 'darwin' || os.platform() === 'linux') {
-        const totalMemoryInGB = os.totalmem() / 2 ** 30;
-
-        // 25% of available ram up to a maximum of 6gb
-        settings.virtualMachine.memoryInGB = Math.min(6, Math.round(totalMemoryInGB / 4.0));
+      if (Object.keys(deploymentProfiles.defaults).length) {
+        _.merge(settings, deploymentProfiles.defaults);
+        if (!_.has(deploymentProfiles.defaults, 'virtualMachine.memoryInGB')) {
+          setDefaultMemory = true;
+        }
+      } else {
+        _isFirstRun = true;
+        setDefaultMemory = true;
       }
     }
+    if (setDefaultMemory && (os.platform() === 'darwin' || os.platform() === 'linux')) {
+      const totalMemoryInGB = os.totalmem() / 2 ** 30;
+
+      // 25% of available ram up to a maximum of 6gb
+      settings.virtualMachine.memoryInGB = Math.min(6, Math.round(totalMemoryInGB / 4.0));
+    }
     if (os.platform() === 'linux' && !process.env['APPIMAGE']) {
       settings.application.updater.enabled = false;
+    } else {
+      const appVersion = getProductionVersion();
+
+      // Auto-update doesn't work for CI or local builds, so don't enable it by default
+      if (appVersion.includes('-') || appVersion.includes('?')) {
+        settings.application.updater.enabled = false;
+      }
     }
+  }
+  _.merge(settings, lockedProfileSettings);
+  save(settings);
+  lockedSettings = determineLockedFields(lockedProfileSettings) as LockedSettingsType;
 
-    const appVersion = getProductionVersion();
+  return settings;
+}
+export function getLockedSettings(): LockedSettingsType {
+  return lockedSettings;
+}
 
-    // Auo-update doesn't work for CI or local builds, so don't enable it by default
-    if (appVersion.includes('-') || appVersion.includes('?')) {
-      settings.application.updater.enabled = false;
-    }
+export function clearLockedSettings() {
+  lockedSettings = {};
+}
 
-    save(settings);
+/**
+ * Returns an object that mirrors `lockedProfileSettings` but all leaves are `true`.
+ * @param lockedProfileSettings
+ */
+export function determineLockedFields(lockedProfileSettings: LockedSettingsType): LockedSettingsType|boolean {
```

Comment:
This should just return `LockedSettingsType`; it looks like the `boolean` part is just an artifact of the recursive implementation. We can present a better API by adjusting how we do recursion:

```ts
export function determineLockedFields(lockedProfileSettings: LockedSettingsType): LockedSettingsType {
  function isLockedSettingsType(input: any): input is LockedSettingsType {
    return typeof input === 'object' && !Array.isArray(input) && input !== null;
  }

  return Object.fromEntries(Object.entries(lockedProfileSettings).map(([k, v]) => {
    return [k, isLockedSettingsType(v) ? determineLockedFields(v) : true];
  }));
}
```

Reply:
done

Reply:
done